wled00/FX.cpp

```diff
@@ -93,7 +94,58 @@ uint16_t mode_static(void) {
 }
 static const char _data_FX_MODE_STATIC[] PROGMEM = "Solid";
 
-
+/*
+ * Copy selected segment 
+ */
+uint16_t mode_copy_segment(void) {
+
+  uint32_t sourceid = SEGMENT.custom1;
+  SEGMENT.fadeToBlackBy(16); // fades out unused pixels, still allows overlay (also fades out if invalid ID is set)
+  if (sourceid >= strip._segments.size() || sourceid == strip.getCurrSegmentId()) return FRAMETIME; // invalid source
+  CRGB sourcecolor;
+  if (strip._segments[sourceid].isActive()) {
+    // note: copying 1D to 2D as well as 2D to 1D is not supported
+    if(SEGMENT.is2D() && strip._segments[sourceid].is2D()) { // 2D setup
+      uint32_t cx, cy; // sizes to copy
+      cx = std::min(strip._segments[sourceid].virtualWidth(), SEGMENT.virtualWidth()); // get smaller width
+      cy = std::min(strip._segments[sourceid].virtualHeight(), SEGMENT.virtualHeight()); // get smaller height
+      for (unsigned x = 0; x < cx; x++) {
+        for (unsigned y = 0; y < cy; y++) {
+          sourcecolor = strip._segments[sourceid].getPixelColorXY(x, y);   
+          if(SEGMENT.custom2 > 0) // color shifting enabled
+          {
+            CHSV pxHSV = rgb2hsv(sourcecolor); //convert to HSV
+            pxHSV.h += SEGMENT.custom2; // shift hue
+            hsv2rgb_spectrum(pxHSV, sourcecolor); // convert back to RGB 
+            
+          }
```

Comment:
This gets copy and pasted below - probably it wants to be a little static function instead.

Reply:
indeed, can you help out? I do not know what the best way to implement it is. just adding a function?

Reply:
oh, I though you meant some fancy C++ macro magic ;) 
but good point, will add this to color utils

Reply:
if the `shift_hue` function is on a critical path (like pixel copy), then it would be smarter to put the static function into fx.cpp so the compiler can inline. Inlined functions are faster (no overhead due to call-and-return). 

But the compiler will not inline things that are defined in a different .cpp file.